java/src/main/java/ai/onnxruntime/OnnxTensor.java

```diff
@@ -345,7 +347,14 @@ static OnnxTensor createTensor(OrtEnvironment env, OrtAllocator allocator, Objec
         }
       } else {
         if (info.shape.length == 0) {
-          data = OrtUtil.convertBoxedPrimitiveToArray(data);
+          data = OrtUtil.convertBoxedPrimitiveToArray(info.type, data);
+          if (data == null) {
+            throw new OrtException(
+                "Failed to convert a boxed primitive to an array, this is an error with ORT itself, please report it. JavaType = "
```

Comment:
>this is an error with ORT itself [](http://example.com/codeflow?start=66&length=32)

Is there a specific message returned from ORT and can it be propagated?

Reply:
Ah, no I mean if you get that exception it's a logic error in the Java API, and they should report it because we added support for a new type and missed this method. It's unlikely to be triggered until Java gets value types and then there will be user created primitives.

Reply:
What I meant, it is hard to report something that has no starting point for investigation on our side, or, perhjaps, given a message, the user may figure it out on their own.

Reply:
If they paste in the stack trace and associated message then I'll know what went wrong. That's why it prints the javatype and data value at the end of the exception message.

Reply:
I updated the message slightly to say it's a bug in the Java API and they should report the exception message.